browser/brave_browser_process_impl.cc

```diff
@@ -98,6 +101,58 @@ void InitSystemRequestHandlerCallback() {
       ->RegisterOnBeforeSystemRequestCallback(before_system_request_callback);
 }
 
+#if BUILDFLAG(ENABLE_BRAVE_REFERRALS)
+class ReferralServiceDelegate : public brave::BraveReferralsService::Delegate,
```

Comment:
I think it would be good to use separate file for this class. Is there any reason why declares here?